week-2-wide-perspective/w02e01-functional-warm-up/src/helpers.js

```diff
@@ -0,0 +1,61 @@
+export const getLength = (/** @type {object[]} */ arr) => {
+  if (!Array.isArray(arr)) {
+    throw new Error('arr must be of the array type');
+  }
+  return arr.length;
+};
+
+export const getByProperty = (/** @type {object} */ obj, /** @type {string[]} */ ...properties) => {
+  let tempObj = obj;
+  if (!tempObj) {
+    throw new Error('Object is required');
+  }
+  if (!properties) {
+    throw new Error('Property is required');
+  }
+  properties.forEach((property) => {
+    if (!tempObj[property] && tempObj[property] !== 0) {
+      throw new Error(`Property ${property} does not exist`);
+    }
+    tempObj = tempObj[property];
+  });
+  return properties.reduce((prev, curr) => prev[curr], obj);
```

Comment:
nie lepiej przenieść od razu logikę z `throw` do tego reducera ?

w sensie, wyżej już raz "przebiegasz" się po obiekcie.
tutaj robisz to 2 raz.

optymalniej byłoby to zrobić w jednym miejscu